lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -979,7 +979,8 @@ private function executeUpdates($class)
             }
 
             if ($preUpdateInvoke !== ListenersInvoker::INVOKE_NONE) {
-                $this->listenersInvoker->invoke($class, Events::preUpdate, $entity, new PreUpdateEventArgs($entity, $this->em, $this->getEntityChangeSet($entity)), $preUpdateInvoke);
+                $changeSet = $this->getEntityChangeSet($entity);
+                $this->listenersInvoker->invoke($class, Events::preUpdate, $entity, new PreUpdateEventArgs($entity, $this->em, $changeSet), $preUpdateInvoke);
```

Comment:
Still one new error there:
> 1) Doctrine\Tests\ORM\Functional\LifecycleCallbackTest::testLifecycleListener_ChangeUpdateChangeSet
> Exception: [Doctrine\ORM\NoResultException] No result was found for query although at least one row was expected.

`getEntityChangeSet()` returns a reference which gets lost during assignment here.

Reply:
Oh sorry! That’s a bug in Psalm. Will fix.

Reply:
Psalm bugfix: https://github.com/vimeo/psalm/commit/7fa541e39bce2423e517bd60b861cf73de522db7
PHPStan ticket: https://github.com/phpstan/phpstan/issues/757